File: java/src/main/java/com/microservices/rentaloffer/Need.java
package com.microservices.rentaloffer;

import java.io.IOException;

public class Need {

    public static void main(String[] args) {
        String host = args[0];
        String port = args[1];

        Need.publish(host, port);
    }

    public static void publish(String host, String port) {
        try (Connections connection = new Connections(host, port)) {
            connection.publish(new NeedPacket().toJson());
        } catch (Exception e) {
            throw new RuntimeException("Could not publish message:", e);
        }
    }
}

File: csharp_v1/RentalOffer.Need/Need.cs
using System;

using RabbitMQ.Client;

using RentalOffer.Core;

namespace RentalOffer.Need { 

    public class Need {
        
        public static void Main(string[] args) {
            string host = args[0];
            string port = args[1];

            new Connection(host, port).WithOpen(new Need().PublishNeed);
        }

        private void PublishNeed(Connection connection)
        {
            Console.WriteLine(" [*] Publishing need on the bus... To exit press CTRL+C");
            while (true)
            {
                string message = new NeedPacket().ToJson();
                connection.Publish(message);
                Console.WriteLine(" [x] Published {0} on the bus", message);
                System.Threading.Thread.Sleep(5000);
            }
        }

    }

}
